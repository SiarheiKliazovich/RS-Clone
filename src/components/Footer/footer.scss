@import "../../sass/variables";
@import "../../sass/mixins";

.footer {
  width: 100%;
  float: left;
  background-repeat: no-repeat;
  background-size: cover;
  background: #4788ff;
  padding: 0;
  position: relative;
}

.footer-wrapper {
  padding-top: 100px;
  padding-bottom: 45px;
}

.footer__columns_section {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.section__logo {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.footer-logo {
  display: inline-block;
  height: 50px;
  width: 200px;
  background-image: url(../../assets/images/png/logo.png);
  background-size: 100% 100%;
  background-position: center center;
  background-repeat: no-repeat;
  margin: 0px 1px 15px;
  cursor: pointer;
}

.section__logo_text {
  font-size: $font-size-xlg;
  font-family: $font-family-base;
  color: white;
}



.footer-title {
  font-size: $font-size-xlg;
  font-family: $font-family-base;
  color: white;
  padding-bottom: 10px;
}

.footer__list_item {
  font-size: $font-size-llg;
  color: #cfc1c1;
  padding-bottom: 10px;
}

.item-link__social_face {
  background: url(../../assets/images/png/face.webp);
}

.item-link__social_twit {
  background: url(../../assets/images/png/twit.png);
}

.item-link__social_inst {
  background: url(../../assets/images/png/inst1.png);
}

.section__contact_link {
  list-style: none;
  color: #cfc1c1;
}

.copyright-footer {
  padding: 30px 1px;
  border-top: 3px solid rgb(255 255 255 / 40%);
}

.copyright-wrapper {
  display: flex;
  padding: 0 30px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.copyright-text {
  font-size: $font-size-xlg;
  font-family: $font-family-base;
  color: white;
}

.copyright-footer__link {
  background-image: url(../../assets/images/png/pngwin.png);
  background-repeat: no-repeat;
  width: 50px;
  height: 50px;
}

.logo-link {
  background-image: url(../../assets/images/svg/rs_school_js.svg);
  background-repeat: no-repeat;
  width: 100px;
  height: 40px;
}