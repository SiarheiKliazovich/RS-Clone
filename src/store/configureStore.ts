/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
