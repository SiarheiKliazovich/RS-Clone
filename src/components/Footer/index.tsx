import { FunctionComponent } from "react";
import "./footer.scss";
import { FooterType } from "../../types/footer";
import { nightTheme } from "../../data/constants";
import { Link } from "react-router-dom";

const Footer: FunctionComponent<FooterType> = ({
  accentColor,
  isNavbarNightMode,
}: FooterType) => {
  const backgroundColor = nightTheme.background.element;

  return (
    <div
      className="footer"
      style={{
        backgroundColor: isNavbarNightMode
          ? backgroundColor
          : accentColor.static,
      }}
    >
      <div className="container">
        <div className="footer-wrapper">
          <div className="footer__columns_section">
            <div className="section__logo">
              <Link to={"/"}>
                <span className="footer-logo"></span>
              </Link>
              <p className="section__logo_text">Thank you for choosing us</p>
              <ul className="footer__section__list">
                <li className="footer__list_item">
                  <Link className="item-link_sci" to=""></Link>
                </li>
                <li className="footer__list_item">
                  <Link className="footer__list_item-link" to=""></Link>
                </li>
                <li className="footer__list_item">
                  <Link className="footer__list_item-link" to=""></Link>
                </li>
              </ul>
            </div>
            <div className="section__menu">
              <h3 className="footer-title">Menu</h3>
              <ul className="footer__section__list">
                <li className="footer__list_item">
                  <Link className="footer__list_item-link" to={"/"}>
                    Home
                  </Link>
                </li>
                <li className="footer__list_item">
                  <Link className="footer__list_item-link" to={"/configurator"}>
                    Configurator
                  </Link>
                </li>
                <li className="footer__list_item">
                  <Link className="footer__list_item-link" to={"/tiemarket"}>
                    Tie Market
                  </Link>
                </li>
              </ul>
            </div>
            <div className="section__contact">
              <h3 className="footer-title">Contact us</h3>
              <ul className="footer__section__list">
                <li className="footer__list_item">
                  <Link className="section__contact_link" to="">
                    +0123456789
                  </Link>
                </li>
                <li className="footer__list_item">
                  <Link className="section__contact_link" to="">
                    <span>TieConfigSupport@.com</span>
                  </Link>
                </li>
                <li></li>
              </ul>
            </div>
          </div>
        </div>
        <div className="copyright-footer">
          <div className="copyright-wrapper">
            <span className="copyright-text">Copyright 2023 by</span>
            <Link className="copyright-footer__link" to=""></Link>
            <Link className="copyright-footer__link" to=""></Link>
            <Link className="copyright-footer__link" to=""></Link>
            <Link className="logo-link" to=""></Link>
          </div>
        </div>
      </div>
    </div>
  );
};
export default Footer;
