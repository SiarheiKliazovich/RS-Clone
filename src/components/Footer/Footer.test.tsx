import { render } from "@testing-library/react";
import Footer from ".";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "../../store/configureStore";

const store = configureStore();

describe("Footer", () => {
  test("renders correctly", () => {
    const container = render(
      <Provider store={store}>
        <Router>
          <Footer
            accentColor={{ static: "#FFF", hover: "#000" }}
            isNavbarNightMode={false}
          />
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
