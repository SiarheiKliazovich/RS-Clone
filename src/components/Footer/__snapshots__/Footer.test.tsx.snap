// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer"
        style="background-color: rgb(255, 255, 255);"
      >
        <div
          class="container"
        >
          <div
            class="footer-wrapper"
          >
            <div
              class="footer__columns_section"
            >
              <div
                class="section__logo"
              >
                <a
                  href="/"
                >
                  <span
                    class="footer-logo"
                  />
                </a>
                <p
                  class="section__logo_text"
                >
                  Спасибо, что выбрали нас
                </p>
                <ul
                  class="footer__section__list"
                >
                  <li
                    class="footer__list_item"
                  >
                    <a
                      class="item-link_sci"
                      href="/"
                    />
                  </li>
                  <li
                    class="footer__list_item"
                  >
                    <a
                      class="footer__list_item-link"
                      href="/"
                    />
                  </li>
                  <li
                    class="footer__list_item"
                  >
                    <a
                      class="footer__list_item-link"
                      href="/"
                    />
                  </li>
                </ul>
              </div>
              <div
                class="section__menu"
              >
                <h3
                  class="footer-title"
                >
                  МЕНЮ
                </h3>
                <ul
                  class="footer__section__list"
                >
                  <li
                    class="footer__list_item"
                  >
                    <a
                      class="footer__list_item-link"
                      href="/"
                    >
                      Главная
                    </a>
                  </li>
                  <li
                    class="footer__list_item"
                  >
                    <a
                      class="footer__list_item-link"
                      href="/configurator"
                    >
                      Конфигуратор
                    </a>
                  </li>
                  <li
                    class="footer__list_item"
                  >
                    <a
                      class="footer__list_item-link"
                      href="/tie-market"
                    >
                      Маркет
                    </a>
                  </li>
                </ul>
              </div>
              <div
                class="section__contact"
              >
                <h3
                  class="footer-title"
                >
                  Наши контакты
                </h3>
                <ul
                  class="footer__section__list"
                >
                  <li
                    class="footer__list_item"
                  >
                    <a
                      class="section__contact_link"
                      href="/"
                    >
                      +0123456789
                    </a>
                  </li>
                  <li
                    class="footer__list_item"
                  >
                    <a
                      class="section__contact_link"
                      href="/"
                    >
                      <span>
                        TieConfigSupport@.com
                      </span>
                    </a>
                  </li>
                  <li />
                </ul>
              </div>
            </div>
          </div>
          <div
            class="copyright-footer"
          >
            <div
              class="copyright-wrapper"
            >
              <span
                class="copyright-text"
              >
                Copyright 2023 by
              </span>
              <a
                class="copyright-footer__link"
                href="https://github.com/siarheikliazovich"
                target="_blank"
              />
              <a
                class="copyright-footer__link"
                href="https://github.com/anast-ananko"
                target="_blank"
              />
              <a
                class="copyright-footer__link"
                href="https://github.com/ablbsk"
                target="_blank"
              />
              <a
                class="logo-link"
                href="https://rs.school"
                target="_blank"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer"
      style="background-color: rgb(255, 255, 255);"
    >
      <div
        class="container"
      >
        <div
          class="footer-wrapper"
        >
          <div
            class="footer__columns_section"
          >
            <div
              class="section__logo"
            >
              <a
                href="/"
              >
                <span
                  class="footer-logo"
                />
              </a>
              <p
                class="section__logo_text"
              >
                Спасибо, что выбрали нас
              </p>
              <ul
                class="footer__section__list"
              >
                <li
                  class="footer__list_item"
                >
                  <a
                    class="item-link_sci"
                    href="/"
                  />
                </li>
                <li
                  class="footer__list_item"
                >
                  <a
                    class="footer__list_item-link"
                    href="/"
                  />
                </li>
                <li
                  class="footer__list_item"
                >
                  <a
                    class="footer__list_item-link"
                    href="/"
                  />
                </li>
              </ul>
            </div>
            <div
              class="section__menu"
            >
              <h3
                class="footer-title"
              >
                МЕНЮ
              </h3>
              <ul
                class="footer__section__list"
              >
                <li
                  class="footer__list_item"
                >
                  <a
                    class="footer__list_item-link"
                    href="/"
                  >
                    Главная
                  </a>
                </li>
                <li
                  class="footer__list_item"
                >
                  <a
                    class="footer__list_item-link"
                    href="/configurator"
                  >
                    Конфигуратор
                  </a>
                </li>
                <li
                  class="footer__list_item"
                >
                  <a
                    class="footer__list_item-link"
                    href="/tie-market"
                  >
                    Маркет
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="section__contact"
            >
              <h3
                class="footer-title"
              >
                Наши контакты
              </h3>
              <ul
                class="footer__section__list"
              >
                <li
                  class="footer__list_item"
                >
                  <a
                    class="section__contact_link"
                    href="/"
                  >
                    +0123456789
                  </a>
                </li>
                <li
                  class="footer__list_item"
                >
                  <a
                    class="section__contact_link"
                    href="/"
                  >
                    <span>
                      TieConfigSupport@.com
                    </span>
                  </a>
                </li>
                <li />
              </ul>
            </div>
          </div>
        </div>
        <div
          class="copyright-footer"
        >
          <div
            class="copyright-wrapper"
          >
            <span
              class="copyright-text"
            >
              Copyright 2023 by
            </span>
            <a
              class="copyright-footer__link"
              href="https://github.com/siarheikliazovich"
              target="_blank"
            />
            <a
              class="copyright-footer__link"
              href="https://github.com/anast-ananko"
              target="_blank"
            />
            <a
              class="copyright-footer__link"
              href="https://github.com/ablbsk"
              target="_blank"
            />
            <a
              class="logo-link"
              href="https://rs.school"
              target="_blank"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
